opendevin/condenser/prompts.py

```diff
@@ -0,0 +1,58 @@
+from opendevin.core.exceptions import (
+    InvalidSummaryResponseError,
+    LLMMalformedActionError,
+    LLMResponseError,
+)
+from opendevin.core.logger import opendevin_logger as logger
+from opendevin.core.utils import json
+from opendevin.events.action.agent import AgentSummarizeAction
+from opendevin.events.event import EventSource
+from opendevin.events.serialization.action import action_from_dict
+
+WORD_LIMIT = 200
+MESSAGE_SUMMARY_WARNING_FRAC = 0.75
+
+SUMMARY_PROMPT_SYSTEM = """
+Your job is to summarize a history of previous messages in a conversation between an AI persona and a human. The conversation you are given is a from a fixed context window and may not be complete. Keep your summary less than {WORD_LIMIT} words, do NOT exceed this word limit.
+Only output the summary, do NOT include anything else in your output.
+Given the following actions and observations, create a JSON response with:
+    - "action": "summarize"
+    - args:
+      - "summarized_actions": A precise sentence summarizing all the provided actions, written in the first person.
+      - "summarized_observations": A few precise sentences summarizing all the provided observations, written in the third person.
```

Comment:
Just curious: what's the motivation behind "first person" for action and "third person" for observations?

Reply:
We still need to do some work on improving the prompt. This was adapted from https://github.com/OpenDevin/OpenDevin/pull/2021. From what I felt, actions are performed by the agent, and it would make more sense to have "first person" for the action summary as this gets passed to the agent in the prompt. Similarly "third person" for observation. 
I didn't pay much attention to it that time, but now that you've pointed it out, maybe we can avoid specifying the person in the new prompt, as this summary message has `role = 'user'` when passed to the llm.

Reply:
My intuition is it doesn't matter at all, thus the ask.

Reply:
CodeAct prompt is structured as a dialogue, one message from user, one from agent pattern. The former version was splitting this result in 2 messages sent to the LLM, following the same pattern. It went like this:
```
role='user'
...The task is ... 

role='agent'
For this task, I did this and I did that and the other...
# action-like

role= 'user'
The code was ... and the assistant had....
# obs-like
```
In this PR, we're sending them both in the same role='user' message. So it should probably be all like an obs. It will also be merged into the user task, e.g. it will end up like:
```
role='user'
The task is... 
The assistant did this and that...
The code was... and...
```